lib/vagrant/action/builtin/is_env_set.rb

```diff
@@ -0,0 +1,23 @@
+module Vagrant
+  module Action
+    module Builtin
+      # This middleware is meant to be used with Call and can check if
+      # a variable in env is set.
+      class IsEnvSet
+        def initialize(app, env, key, **opts)
+          @app    = app
+          @logger = Log4r::Logger.new("vagrant::action::builtin::is_env_set")
+          @key    = key
+          @invert = !!opts[:invert]
```

Comment:
Maybe I'm missing something, but I'm not seeing where this is used? Could you document it?